ui/src/app/shared/models.ts

```diff
@@ -283,7 +283,7 @@ export interface Info {
 
 export interface ApplicationSpec {
     project: string;
-    source: ApplicationSource;
+    source?: ApplicationSource;
     sources: ApplicationSource[];
```

Comment:
Shall this also be optional?

Reply:
I think so, otherwise the YAML version doesn't correspond to the create new app configuration as it would have to include an empty _source field_. As far as I understood the soucre field has also been deprecated in https://github.com/argoproj/argo-cd/blob/master/docs/proposals/multiple-sources-for-applications.md#add-new-sources-field-in-application-spec

Reply:
Sorry, I meant the `sources` field, in case we create a single-source app.

Reply:
I see.. in that case, the implementation still uses the `sources` field containing a list with a single source, but you're right: the user might still wish to use the `source` field when editing it in YAML mode. Shall we allow it? Or rather show a _deprecated_ message?

Reply:
I may not know all details how it works now. If that's the case, let's keep it as is.

Reply:
Alright